vllm/config/compilation.py

```diff
@@ -109,11 +109,50 @@ class PassConfig:
     """Whether to enable async TP."""
     enable_fi_allreduce_fusion: bool = False
     """Whether to enable flashinfer allreduce fusion."""
-    fi_allreduce_fusion_max_token_num: int = 16384
-    """Max number of tokens to used in flashinfer allreduce fusion."""
+    fi_allreduce_fusion_max_size_mb: dict[int,
+                                          float] = field(default_factory=dict)
+    """The thresholds of the communicated tensor sizes under which
+    vllm should use flashinfer fused allreduce. Specified as a
+    dictionary mapping each world size to the threshold in MB
+        { <world size>: <max size in mb> }
+    Unspecified world sizes will fallback to
+        _FI_ALLREDUCE_MAX_INPUT_SIZES = {
+            "9.0": {
+                2: 64 * MiB,  # 64MB
+                4: 2 * MiB,  # 2MB
+                8: 1 * MiB,  # 1MB
+            },
+            "10.0": {
+                2: 64 * MiB,  # 64MB
+                4: 32 * MiB,  # 32MB
+                8: 1 * MiB,  # 1MB
+            },
+        }, where key is the device capability"""
```

Comment:
Let's set the default dict to `_FI_ALLREDUCE_MAX_INPUT_SIZES` and then in `__post_init__` we can do:
```
self.fi_allreduce_fusion_max_size_mb = {**_FI_ALLREDUCE_MAX_INPUT_SIZES, **fi_allreduce_fusion_max_size_mb}
```

cc @hmellor would this work? Or should we just generate this docstring from `_FI_ALLREDUCE_MAX_INPUT_SIZES`?

Reply:
I am also realizing that `FI_ALLREDUCE_FUSION_MAX_SIZE_MB` has different default for different compute capabilities - so we can't set it as the default for the config.